server/plugin.go

```diff
@@ -661,3 +677,26 @@ func (p *Plugin) getUsername(mmUserID string) (string, error) {
 
 	return "@" + info.BitbucketUsername, nil
 }
+
+func (p *Plugin) getURL(selfHostedURL, defaultURL string) string {
+	if selfHostedURL != "" {
+		return selfHostedURL
+	}
+	return defaultURL
+}
+
+// getBitbucketBaseURL returns the Bitbucket Server URL from the configuration
+// if there is a Self Hosted URL configured it returns it
+// if not it will return the Bitbucket Cloud base URL
+func (p *Plugin) getBitbucketBaseURL() string {
+	config := p.getConfiguration()
+	return p.getURL(config.BitbucketSelfHostedURL, BitbucketBaseURL)
+}
+
+// getBitbucketAPIBaseURL returns the Bitbucket Server API URL from the configuration
+// if there is a API Self Hosted URL configured it returns it
+// if not it will return the Bitbucket Cloud API base URL
+func (p *Plugin) getBitbucketAPIBaseURL() string {
+	config := p.getConfiguration()
+	return p.getURL(config.BitbucketAPISelfHostedURL, BitbucketAPIBaseURL)
+}
```

Comment:
I think this de-duplication actually makes this a bit more difficult to read. What are your thoughts on this instead?

```go
config := p.getConfiguration()
if config.BitbucketSelfHostedURL != "" {
    return config.BitbucketSelfHostedURL
}

return BitbucketBaseURL
```

Note that part of it is that `getURL` doesn't really have anything to do with URLs, and probably shouldn't be defined on the `Plugin` struct. I know that's already implemented that way with `getBaseURL` though

Reply:
Makes sense! Done!